src/Symfony/Component/Security/Http/EntryPoint/FormAuthenticationEntryPoint.php

```diff
@@ -54,7 +54,7 @@ public function start(Request $request, AuthenticationException $authException =
 
             $response = $this->httpKernel->handle($subRequest, HttpKernelInterface::SUB_REQUEST);
             if (200 === $response->getStatusCode()) {
-                $response->headers->set('X-Status-Code', 401);
+                $response->setStatusCode(401);
```

Comment:
@fabpot interested on hearing your thoughts on the above? would be good to get this into 3.2 before the development window closes